test/test_bson.py

```diff
@@ -58,6 +58,21 @@
     long = int
 
 
+class StrictStringIO(object):
```

Comment:
Can we add a new test with your repro (ie using a real file returned by `open()`)? That would help catch future changes to Python's api.

Reply:
Done.  I've also turned the test into a more Go-like loop-based test to reduce repetition.

Reply:
A couple follow-up thoughts:

- Would it be worth returning the `StrictStringIO` tests to just use `StringIO`?
- Or maybe get rid of them entirely?

Reply:
Thanks! Yeah I think we can remove `StrictStringIO` and keep the `StringIO` ones as they were before.

Reply:
Done. Also fixed the formatting issues I introduced (so used to 2-space indents…).

Also added a message to the `assertRaises`es so that if it does fail it tells which input caused it.